lib/buffer.js

```diff
@@ -246,18 +247,30 @@ function fromArrayLike(obj) {
   const length = obj.length;
   const b = allocate(length);
   for (var i = 0; i < length; i++)
-    b[i] = obj[i] & 255;
+    b[i] = obj[i];
   return b;
 }
 
 function fromArrayBuffer(obj, byteOffset, length) {
+  if (!isArrayBuffer(obj))
+    throw new TypeError('argument is not an ArrayBuffer');
```

Comment:
Basically, this error can only be raised if someone’s messing with the prototype of `obj` to make it look like an `ArrayBuffer`, right?

If the others in @nodejs/buffer agree I’d suggest dropping this check then.